File: 1332.py
class Solution:
    def removePalindromeSub(self, s: str) -> int:
        if not s:
            return 0
        half = len(s) // 2
        if s[:half] == s[-1:-1-half:-1]:
            return 1
        return 2


if __name__ == "__main__":
    s = Solution()
    print()

File: 1332.go
package leetcode

func removePalindromeSub(s string) int {
	n := len(s)
	for i := 0; i < n/2; i++ {
		if s[i] != s[n-i-1] {
			return 2
		}
	}
	return 1
}
